### DISCUSSION TITLE
Using kwok docker image with docker-compose

### DISCUSSION THREAD

**Discussion by orudi** [kubernetes-sigs/kwok#399] [Q&A] [answered]
Hi!
What would be a good way to use kwok docker image with docker-compose and have an ability to call kwok functionality from other containers? I have not discovered if kwok exposes some network api (rest, socket), so the only thing that comes to mind is to use sshd. So I tried to use custom Dockerfile 

> FROM registry.k8s.io/kwok/cluster:v1.26.0
> 
> RUN apk add --update --no-cache openrc openssh
> RUN echo 'PasswordAuthentication yes' >> /etc/ssh/sshd_config
> RUN echo 'PermitRootLogin yes' >> /etc/ssh/sshd_config
> RUN echo 'root:SophisticatedPassword42' | chpasswd
> 
> ENTRYPOINT ["/entrypoint.sh"]
> EXPOSE 22
> COPY entrypoint.sh /
> 

and I also took the entrypoint file from base docker image and added 2 lines at the start to call sshd in the backgroud

> ssh-keygen -A
> /usr/sbin/sshd -D -e &

The resulted docker image starts, and is able to accept ssh session. But then I discovered that there is an error with cluster creation

> 053b0b459174:~# kwokctl create cluster --name kl_2
> Creating cluster                                                                                                                       cluster="kl_2"
> Pull image                                                                                        cluster="kl_2" image="registry.k8s.io/etcd:3.5.6-0"
> ERROR Failed to setup config      cluster="kl_2" err="docker pull registry.k8s.io/etcd:3.5.6-0: exec: \"docker\": executable file not found in $PATH"
> Cluster is cleaned up                                                                                                                  cluster="kl_2"
> ERROR Execute exit                               err="docker pull registry.k8s.io/etcd:3.5.6-0: exec: \"docker\": executable file not found in $PATH"
> 053b0b459174:~#

So something is not correct wiht this setup.

**Comment by wzshiming** [comment, marked as answer]
https://github.com/kubernetes-sigs/kwok/blob/64d08be57603042e85e330c74d1c880782c3b404/images/cluster/Dockerfile#L43-L49

These env variables should be needed to allow kwokctl to use binary runtime

**Comment by wzshiming** [comment]
https://github.com/kubernetes-sigs/kwok/issues/261

Exposing the ports of components will be supported in the next release
